tests/e2e/transaction_nonActor_apis/hellodapr_test.go

```diff
@@ -0,0 +1,193 @@
+// +build e2e
+
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package hellodapr_e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/dapr/dapr/tests/e2e/utils"
+	kube "github.com/dapr/dapr/tests/platforms/kubernetes"
+	"github.com/dapr/dapr/tests/runner"
+	"github.com/stretchr/testify/require"
+)
+
+type testCommandRequest struct {
+	Message string `json:"message,omitempty"`
+}
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+const numHealthChecks = 60 // Number of times to check for endpoint health per app.
+
+var tr *runner.TestRunner
+
+func TestMain(m *testing.M) {
+	// This test shows how to deploy the multiple test apps, validate the side-car injection
+	// and validate the response by using test app's service endpoint
+
+	// These apps will be deployed for hellodapr test before starting actual test
+	// and will be cleaned up after all tests are finished automatically
+	testApps := []kube.AppDescription{
+		{
+			AppName:        "hellobluedapr",
+			DaprEnabled:    true,
+			ImageName:      "e2e-hellodapr",
+			Replicas:       1,
+			IngressEnabled: true,
+		},
+	}
+
+	tr = runner.NewTestRunner("hellodapr", testApps, nil, nil)
+	os.Exit(tr.Start(m))
+}
+
+var helloAppTests = []struct {
+	in               string
+	app              string
+	testCommand      string
+	expectedResponse string
+}{
+	{
+		"green dapr",
+		"hellogreendapr",
+		"green",
+		"Hello green dapr!",
+	},
+	{
+		"blue dapr",
+		"hellobluedapr",
+		"blue",
+		"Hello blue dapr!",
+	},
+}
+
+func generateTestCases() []testCase {
+	// Just for readability
+	emptyRequest := requestResponse{
+		nil,
+	}
+
+	// Just for readability
+	emptyResponse := requestResponse{
+		nil,
+	}
+
+	testCase1Key := "daprsecret"
+	testCase1Value := "admin"
+
+	return []testCase{
+		{
+			// No comma since this will become the name of the test without spaces.
+			"Test get with empty request response for single app and single hop",
+			[]testStep{
+				{
+					"get",
+					emptyRequest,
+					emptyResponse,
+				},
+			},
+		},
+		{
+			// No comma since this will become the name of the test without spaces.
+			"Test get a single item for single app and single hop",
+			[]testStep{
+				{
+					"get",
+					newRequest(utils.SimpleKeyValue{testCase1Key, testCase1Value}),
+					newResponse(utils.SimpleKeyValue{testCase1Key, testCase1Value}),
+				},
+			},
+		},
+	}
+}
+
+func sendToTransaction(t *testing.T, publisherExternalURL string, topic string) ([]string, error) {
+	var sentMessages []string
```

Comment:
Please add test case to https://github.com/dapr/dapr/blob/master/tests/e2e/stateapp/stateapp_test.go instead of creating new files. this is the part of state api building block.